src/web-server/routes.js

```diff
@@ -13,29 +13,24 @@ export default (workbox, webServer) => {
     workbox.routing.registerRoute(
         wwwRegex,
         async ({ url }) => {
+            const formatter =
+                url.searchParams.get('json') === 'true'
+                    ? jsonFormatter
+                    : htmlFormatter;
             const path = url.pathname.match(wwwRegex)[1];
-            let body;
-            let type;
-            let status;
+
+            let res;
             try {
-                const result = await webServer.serve(path);
-                body = result.body;
-                type = result.type;
-                status = 200;
+                res = await webServer.serve(path, formatter);
             } catch (err) {
-                body = format404(path);
-                type = 'text/html';
-                // TODO: should probably do a better job here on mapping to err
-                status = 404;
+                res = formatter.format404(path);
             }
-
-            const init = {
-                status,
+            const config = {
+                status: res.status,
                 statusText: 'OK',
```

Comment:
If we return a `404`, does `OK` still make sense here?  Or should we move that onto each object too, and let the formatter pick?

Reply:
`OK` is overwritten for anything that is not 200.